src/components/Navigation/Logout.jsx

```diff
@@ -10,7 +11,17 @@ class Logout extends React.Component {
   }
 
   logout() {
-    this.props.dispatch(logout())
+    const application = this.props.app
+    const section = this.props.section.section
+    const subsection = this.props.section.subsection
+
+    saveSection(application, section, subsection, this.props.dispatch)
+      .then(() => {
+        this.props.dispatch(logout())
```

Comment:
This pattern is used around the app, but is sort of backwards. Typically, one would want to have a dispatch handler pass the dispatch to an action, and that action would handle calling whatever functions it needed to. 

For example, in this component:

```
mapDispatchToProps = (dispatch) => ({ handleLogout: authenticationActions.logout(dispatch) })
```

Then after the `saveSection` call: `saveSection(...args).then(handleLogout)`, etc.

In this app, we usually see the pattern of the dispatch being manually passed down through parent props, and then dispatching actions directly from the component. I'm not sure if we want to start transitioning these components to use the typical redux pattern, or if its better to get these bug fixes in and open new issues to audit this code.

Reply:
I think get the bug fixes in, get automated browser tests going, then start doing across the board code audits to refactor things in the typical pattern, to ease maintenance and onboarding.